src/components/ImageCarousel.tsx

```diff
@@ -1,56 +1,75 @@
-import React, { ReactElement, useState } from 'react'
-import { StyleSheet, useWindowDimensions } from 'react-native'
-import Carousel, { Pagination } from 'react-native-snap-carousel'
-import { ImagesType, ImageType } from '../constants/endpoints'
+import React, { ReactElement } from 'react'
+import { ImagesType } from '../constants/endpoints'
+import { useWindowDimensions } from 'react-native'
+import { Pagination } from 'react-native-snap-carousel'
 import styled from 'styled-components/native'
+import ImageViewer from 'react-native-image-zoom-viewer'
+import { COLORS } from '../constants/colors'
+
+const ImageView = styled.View`
+  height: 35%;
+`
 
 const StyledImage = styled.Image`
-  width: 100%;
   height: 100%;
-  position: relative;
 `
 
-const Container = styled.View`
-  height: 35%;
+const PaginationView = styled.View`
+  position: absolute;
+  left: 0;
+  right: 0;
+  top: 10px;
 `
 
-const styles = StyleSheet.create({
-  container: {
-    position: 'absolute',
-    marginTop: -10,
-    alignSelf: 'center'
-  }
-})
-
 interface ImageCarouselPropsType {
   images: ImagesType
 }
 
 interface ItemType {
-  item: ImageType
+  source: {
+    uri: string
+  }
+}
+
+interface ImageUrlType {
+  url: string
 }
 
 const ImageCarousel = ({ images }: ImageCarouselPropsType): ReactElement => {
-  const [activeImage, setActiveImage] = useState(0)
-  const { width: viewportWidth } = useWindowDimensions()
+  const imagesUrls: ImageUrlType[] = images.map(image => ({
+    url: image.image
+  }))
+
+  const { height: viewportHeight } = useWindowDimensions()
+
+  const renderIndicator = (currentIndex?: number, allSize?: number): JSX.Element => {
```

Comment:
Just personal preference.

```suggestion
  const renderIndicator = (currentIndex?: number, allSize?: number): ReactElement => {
```